diff --git a/edit.c b/edit.c
index a367f16..5b42891 100644
--- a/edit.c
+++ b/edit.c
@@ -304,7 +304,11 @@ static void close_pipe(FILE *pipefd)
 	if (WIFSIGNALED(status))
 	{
 		int sig = WTERMSIG(status);
-		if (sig != SIGPIPE || ch_length() != NULL_POSITION)
+		if (
+#ifdef SIGPIPE
+      sig != SIGPIPE ||
+#endif
+      ch_length() != NULL_POSITION)
 		{
 			parg.p_string = signal_message(sig);
 			error("Input preprocessor terminated: %s", &parg);
