diff --git a/charset.c b/charset.c
index 17ead6b..1d22848 100644
--- a/charset.c
+++ b/charset.c
@@ -449,6 +449,12 @@ static void set_charset(void)
 	 * rather than from predefined charset entry.
 	 */
 	ilocale();
+#ifdef __MVS__
+/*
+	 * z/OS Locale does not properly support UTF-8. Default to "utf-8".
+*/
+  (void) icharset("utf-8", 1);
+#endif
 #else
 #if MSDOS_COMPILER
 #if MSDOS_COMPILER==WIN32C
